### DISCUSSION TITLE
Redirect `has` field: check if header or cookie is NOT defined

### DISCUSSION THREAD

**Discussion by bradbarrow** [vercel/next.js#25507] [Help]
I'm trying to understand if it's possible to match a redirect only in the **absence** of a particular header/cookie.

i.e. 

```
async redirects() {
  return [
    {
      source: '/:path*',
      has: {
        type: 'header',
        key: 'x-dont-want-this',
        value: null,
      },
      destination: '/other-path',
    },
  ];
}
```

The idea being that only if the incoming request has NO `x-dont-want-this` header at all, will it be redirected.

I've tried checking a value of `null`, `''`, etc. I've also tried regex's that don't match any values I expect, but in all of those cases I still think the matcher is expecting the header to be defined.

I'm looking for a `hasNot` field, I guess.

Thoughts?

**Comment by dlaugharnTAG** [comment]
To my knowledge you can't do this yet, but I wrote up a feature request here: https://github.com/vercel/next.js/discussions/24650
